### PR TITLE
Todo app on functional components

### PR DESCRIPTION
**Pull request by Vovi4**

### CODE REVIEW

File: src/components/Context/Context.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+export const statusChange = (updatedItem) => ({ type: ACTIONS.STATUS_CHANGE, updatedItem });
+export const deleteTodo = (filteredItems) => ({ type: ACTIONS.DELETE_TODO, filteredItems });
+export const inputHandle = (currentItem) => ({ type: ACTIONS.INPUT_HANDLE, currentItem });
+export const addTodo = (items) => ({ type: ACTIONS.ADD_TODO, items });
+
+export const TodoProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  const contextValue = useMemo(() => {
```
Comment (TarasProtsyshyn) [inline src/components/Context/Context.js]: Why are you need here `useMemo`?

File: src/components/Date/CurrentDate.js [hunk 2]
```diff
@@ -1,25 +1,19 @@
...
+  const current = date.toLocaleString('en-us',{year: 'numeric', month: 'short', day: 'numeric', weekday: 'short'});
 
-    render() {
-        return (
-            <div className="curent-data">
-                {this.state.date}
-            </div>
-        );
-    }
+  const [currentDate, setCurrentData] = useState(current);
```
Comment (TarasProtsyshyn) [inline src/components/Date/CurrentDate.js]: Why do you need here `useState`?

File: src/components/TodoItems/TodoItems.js [hunk 3]
```diff
@@ -1,47 +1,72 @@
...
+  const { state, dispatch } = useContext(TodoContext);
 
-    componentDidUpdate (prevProps) {        
-        if (this.props.entries.length > prevProps.entries.length) {
-            const addedElement = this.props.entries
-            console.log(`New todo ${addedElement[addedElement.length-1].text} was added`)
-            localStorage.setItem("Todo-item", JSON.stringify(this.props.entries))
+  useEffect(() => {
+    localStorage.setItem("Todo-item", JSON.stringify(state.items));
+  });
```
Comment (TarasProtsyshyn) [inline src/components/TodoItems/TodoItems.js]: Why do you don't set any of the dependencies in `useEffect`? Is it's working correctly?

File: src/components/TodoItems/TodoItems.js [hunk 4]
```diff
@@ -1,47 +1,72 @@
...
+
+  const { state, dispatch } = useContext(TodoContext);
 
-    componentDidUpdate (prevProps) {        
-        if (this.props.entries.length > prevProps.entries.length) {
-            const addedElement = this.props.entries
-            console.log(`New todo ${addedElement[addedElement.length-1].text} was added`)
-            localStorage.setItem("Todo-item", JSON.stringify(this.props.entries))
+  useEffect(() => {
+    localStorage.setItem("Todo-item", JSON.stringify(state.items));
```
Comment (TarasProtsyshyn) [inline src/components/TodoItems/TodoItems.js]: Better to do it in `InputTodo` on `handleInput`

File: src/components/Context/Context.js (hunk 1 above)
Comment (Vovi4) [reply src/components/Context/Context.js]: I had doubts about this part of the code, but now I understand that it is unnecessary

File: src/components/Date/CurrentDate.js (hunk 2 above)
Comment (Vovi4) [reply src/components/Date/CurrentDate.js]: Delete state from this component

File: src/components/TodoItems/TodoItems.js (hunk 3 above)
Comment (Vovi4) [reply src/components/TodoItems/TodoItems.js]: You are right its works incorrectly. Without dependencies, the useEffect works on each renderer.

File: src/components/TodoItems/TodoItems.js (hunk 4 above)
Comment (Vovi4) [reply src/components/TodoItems/TodoItems.js]: Thanks for the advice. Moved the addition of item the localStorage in the handleInput function

**Review by TarasProtsyshyn — APPROVED** [review]

**Pull request merged**